addons/test_mail/tests/test_performance.py

```diff
@@ -832,6 +832,38 @@ def test_message_format(self):
             for message in res:
                 self.assertEqual(len(message['attachment_ids']), 2)
 
+    @mute_logger('odoo.tests', 'odoo.addons.mail.models.mail_mail', 'odoo.models.unlink')
+    @users('emp')
+    @warmup
+    def test_message_format_query(self):
+        name_field = self.env['ir.model.fields']._get(self.container._name, 'name')
+        customer_id_field = self.env['ir.model.fields']._get(self.container._name, 'customer_id')
+
+        record1 = self.env['mail.test.simple'].create({'name': 'Test1'})
+        record2 = self.env['mail.test.simple'].create({'name': 'Test2'})
+        record3 = self.env['mail.test.track'].create({'name': 'Test3'})
+
+        messages = self.env['mail.message'].sudo().create([{
```

Comment:
Maybe
```
messages = self.env['mail.message'].sudo().create([{
    'model': record._name,
    'res_id': record.id
} for record in [record1, record2, record3])
```

Reply:
A loop for 3 (explicit) records ? Why not, it doesn't really matter I guess.